numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1558,6 +1570,35 @@ pack_inner(const char *inptr,
             inptr += 16;
         }
     }
+#elif defined __ARM_NEON || defined __ARM_NEON__
+    if (in_stride == 1 && element_size == 1 && n_out > 2) {
+        /* don't handle non-full 8-byte remainder */
+        npy_intp vn_out = n_out - (remain ? 1 : 0);
+        vn_out -= (vn_out & 1);
+        for (index = 0; index < vn_out; index += 2) {
+            unsigned int r;
+            npy_uint64 a = *(npy_uint64*)inptr;
+            npy_uint64 b = *(npy_uint64*)(inptr + 8);
+            if (order == 'b') {
+                a = npy_bswap8(a);
+                b = npy_bswap8(b);
+            }
+            const npy_uint64 __attribute__((aligned(16))) data[2] = {a, b};
+            uint64x2_t v = vld1q_u64(data);
+            uint64x2_t zero = vdupq_n_u64(0);
+            /* false -> 0x00 and true -> 0xFF (there is no cmpneq) */
+            v = vreinterpretq_u64_u8(vceqq_u8(vreinterpretq_u8_u64(v), vreinterpretq_u8_u64(zero)));
+            v = vreinterpretq_u64_u8(vceqq_u8(vreinterpretq_u8_u64(v), vreinterpretq_u8_u64(zero)));
```

Comment:
I think it should be better if you inverted the returned mask via `vmvnq_u8` instead of having two tests.

Reply:
will modify later using `vmvnq_u8`

Reply:
@seiko2plus failed with `vmvnq_u8`, because this is not a `bitwise NOT`  operation, the corresponding sse operation is `_mm_cmpeq_epi8`.

Reply:
Maybe, I wasn't clear enough,=. what I mean is replacing the second test with `vmvnq_u8`, for instance
```C
v = vreinterpretq_u64_u8(vmvnq_u8(vceqq_u8(vreinterpretq_u8_u64(v), vreinterpretq_u8_u64(zero)))); 
````

Reply:
Thanks for the advice, performance improved. 😄